src/Raven.Server/Documents/Handlers/IndexHandler.cs

```diff
@@ -571,7 +571,7 @@ public async Task SetLockMode()
 
                 foreach (var name in parameters.IndexNames)
                 {
-                    await Database.IndexStore.SetLock(name, parameters.Mode);
+                    await Database.IndexStore.SetLock(name, parameters.Mode, string.Empty); // TODO: can't use single guid here for multiple commands
```

Comment:
This is actually a good _reason_ to use a string for the unique id.
You can append to it `id + "/" + i`

Reply:
Nice idea! keep the uniqueness and the injective property.